src/playbacks/hls/hls.js

```diff
@@ -335,12 +335,34 @@ export default class HLS extends HTML5VideoPlayback {
         this.stop()
       }
     } else {
+      // Transforms HLSJS.ErrorDetails.KEY_LOAD_ERROR non-fatal error to
+      // playback fatal error if triggerFatalErrorOnResourceDenied playback
+      // option is set. HLSJS.ErrorTypes.KEY_SYSTEM_ERROR are fatal errors
+      // and therefore already handled.
+      if (this.options.playback
+        && this.options.playback.triggerFatalErrorOnResourceDenied
```

Comment:
Shouldn't we explicity set this to false as default at some init code? do we do already that @clappr/core for other options?

Reply:
It seems that not every options are initialized. Sometimes it uses a getter. This new option is supposed to be a bool value, it's the main reason it is not initialized. I you prefer have it initialized to false in constructor, just tell me and i add the fix.

Reply:
I agree with you =) let's see if the other think the same way and maybe we hack this `feature` into this PR lol

Reply:
We usually don't initialize those nested options to keep initialization simple. What we could do is to always have `this.options.playback` initialized to avoid having to check whether it is defined whenever we want to access it.

@kslimani @leandromoreira, what do you think?

Reply:
Maybe we could have this notion of `option` as first class citizen where we could initialize the defaults:

```javscript

class Option
   const defaults = {
      option: 1,
      hls: {
         option1: 1
      }
   }
   function init()
end

```

It'd make easy to document/read and to centralize any kind of option's behavior.

Reply:
Or maybe we should create a PR for this lol

Reply:
The `player` component is nice place to see how nicely it works for documentation https://github.com/clappr/clappr/blob/dev/src/components/player.js#L153

Reply:
@kslimani I'd avoid creating a first-class citizen for the moment since we'll probably split the options that are visible for each component. We still need to think whether we'll need another component responsible for that or what components will hold references to the options set.

Reply:
@leandromoreira @towerz initializing _[scope]_.options.playback to empty object if undefined is a good idea. Assuming this object is defined allow to simplify many conditions. I too think this initialization could be done in player component, or delegated to core, but in another PR. _(this new PR could introduce the initialization and update other classes conditions where applicable)_

Reply:
In fact, it seems playback option object is [already initialized](https://github.com/clappr/clappr/blob/dev/src/components/player.js#L240).

Reply:
@kslimani indeed it is... we made that to merge compatible options that were added without the `playback` scope. We intend to remove it for `0.4.x` where we'll have a bunch of breaking changes in a few weeks.